### PR TITLE
Fixes #4314

### LINKED ISSUE #4314: Chameleon Gene Doesn't Work For Vampires
Be vampire, get chameleon, nothing happens. They remain completely visible even when not moving. Possibly related to the fact that vampires can get shadowstep which alters their transparency, but I didn't have it at the time and I'm not a coder so I don't really know.

### PR DESCRIPTION
**Pull request by DeityLink**
hahaha, I'm a genius

shadowfix

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life/handle_alpha.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//Refer to life.dm for caller
+
+/mob/living/carbon/human/proc/handle_alpha()
+	..()
+	if(alphas.len < 1)
+		alpha = 255
+	else
+		var/lowest_alpha = 255
+		for(var/alpha_modification in alphas)
+			if(alphas[alpha_modification] < lowest_alpha)
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/life/handle_alpha.dm]: you can do min(alphas[alpha_modification],lowest_alpha)

File: code/modules/spells/targeted/ethereal_jaunt.dm [hunk 2]
```diff
@@ -68,7 +68,7 @@
 		SM.silence_spells(0)
 	target.flags &= ~INVULNERABLE
 	target.incorporeal_move = previncorp
-	target.alpha = prevalpha
+	target.alphas["etheral_jaunt"] = 255
```
Comment (clusterfack) [inline code/modules/spells/targeted/ethereal_jaunt.dm]: probably better to just remove it from the list than this, same with chameleon when it gets deactivated as a gene

File: code/game/dna/genes/goon_powers.dm [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
 	deactivate(var/mob/M, var/connected, var/flags)
 		if(..(M,connected,flags))
-			M.alpha=255
+			M.alphas -= "chameleon_stealth"
```
Comment (N3X15) [inline code/game/dna/genes/goon_powers.dm]: did you test this

File: code/modules/mob/living/carbon/human/life/handle_alpha.dm [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//Refer to life.dm for caller
+
+/mob/living/carbon/human/proc/handle_alpha()
+	..()
```
Comment (N3X15) [inline code/modules/mob/living/carbon/human/life/handle_alpha.dm]: >declaring proc
> LET'S CALL THE PARENT PROC

File: code/game/dna/genes/goon_powers.dm (hunk 3 above)
Comment (DeityLink) [inline code/game/dna/genes/goon_powers.dm]: yeah it works.

File: code/modules/mob/living/carbon/human/life/handle_alpha.dm (hunk 4 above)
Comment (DeityLink) [inline code/modules/mob/living/carbon/human/life/handle_alpha.dm]: >hahaha, I am a genius
>OH NO

File: code/modules/mob/living/carbon/human/life/handle_alpha.dm [hunk 5]
```diff
@@ -0,0 +1,10 @@
+//Refer to life.dm for caller
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/life/handle_alpha.dm]: This file can be removed and the proc moved to mob.dm

File: code/modules/spells/targeted/ethereal_jaunt.dm [hunk 6]
```diff
@@ -44,7 +43,8 @@
 	target.incorporeal_move = 3
 	target.invisibility = INVISIBILITY_MAXIMUM
 	target.flags |= INVULNERABLE
-	target.alpha = 125 //Spoopy mode to know you are jaunting
+	target.alpha = 125
```
Comment (clusterfack) [inline code/modules/spells/targeted/ethereal_jaunt.dm]: can remove this line and put handle_alpha() just below where you add jaunting to alphas

**Comment by clusterfack** [comment]
Can you squash this? Otherwise its good to go

**Comment by DeityLink** [comment]
done

**Pull request merged**